docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_webhook.go

```diff
@@ -46,6 +46,8 @@ Then, we set up the webhook with the manager.
 func (r *CronJob) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithValidator(r).
+		WithDefaulter(r).
```

Comment:
So, before this change, with the deprecated webhook interface `For` was enough to enable both the validation and defaulting logic, if `r` implemented either (there are runtime checks [here](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/builder/webhook.go#L201-L203), if I am not mistaken). 
For custom validator and custom defaulter, you need to specifically `WithValidator` or `WithDefaulter` to enable them.

A test for just the CustomDefaulter is [here](https://github.com/kubernetes-sigs/kubebuilder/pull/3723/files#diff-da5765d58e09cb0d9da8b09d04dde3e3762263e553a8b40e65babc36062af21fR35) and for CustomValidator [here](https://github.com/kubernetes-sigs/kubebuilder/pull/3723/files#diff-d253c7d3da23ab04bf8b977cba0892853e326975afc2938086c8fb9e8130d62bR35).

Reply:
But seems that we are saying here.
The implementation is For the struct `r`
and with the validator using the same struc.
Should we not have a new type, struct for the validation?

Reply:
See: https://github.com/kubernetes-sigs/controller-runtime/blob/4000e996a202917ad7d40f02ed8a2079a9ce25e9/pkg/builder/webhook.go#L70-L74

I understand that you pass an struct representing the Validator and not the object that will be validated.

Reply:
We need the validator struct 
We should not be passing :

```
WithValidator(r).
WithDefaulter(r).
```

Reply:
Agree with Camilla. CustomValidator and CustomDefaulter should be implemented on a separate struct.

Otherwise there would have been no point to introducing these if we still implement them on the API type.

(the goal is to have CustomValidator/CustomDefaulter outside of the api package to reduce the dependencies to controller-runtime in the API package)

Reply:
Thank you to confirm my understanding about the requirements for the changes